File: src/main/java/rails/game/RailsOwnableItem.java
package rails.game;

import rails.game.model.PortfolioModel;
import rails.game.state.Ownable;
import rails.game.state.OwnableItem;

/**
 * RailsOwnableItem is the rails specific version of RailsOwnableItem
 */

public class RailsOwnableItem<T extends Ownable> extends OwnableItem<T> implements RailsItem {

    protected RailsOwnableItem(RailsItem parent, String id, Class<T> type) {
        super(parent, id, type);
    }

    @Override
    public RailsItem getParent() {
        return (RailsItem)super.getParent();
    }
    
    @Override
    public RailsRoot getRoot() {
        return (RailsRoot)super.getRoot();
    }

    /**
     * Moves the item to the owner of the portfolioModel
     * @param model the model of the new owner 
     */
    public void moveTo(PortfolioModel model) {
        moveTo(model.getParent());
    }
}

File: dotnet/RailsLib.Net/Net/Game/RailsOwnableItem.cs
using GameLib.Net.Game.Model;
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Text;

/**
 * RailsOwnableItem is the rails specific version of RailsOwnableItem
 */

namespace GameLib.Net.Game
{
    public class RailsOwnableItem<T> : OwnableItem<T>, IRailsItem where T : IOwnable
    {
        protected RailsOwnableItem(IRailsItem parent, string id /*, Class<T> type */) : base(parent, id)
        {
        }

        IRailsItem IItem<IRailsItem, RailsRoot>.Parent => (IRailsItem)base.Parent;

        RailsRoot IItem<IRailsItem, RailsRoot>.GetRoot => (RailsRoot)base.GetRoot;

        new public IRailsItem Parent
        {
            get
            {
                return (IRailsItem)base.Parent;
            }
        }

        new public RailsRoot GetRoot
        {
            get
            {
                return (RailsRoot)base.GetRoot;
            }
        }

        /**
         * Moves the item to the owner of the portfolioModel
         * @param model the model of the new owner 
         */
        public void MoveTo(PortfolioModel model)
        {
            MoveTo(model.Parent);
        }
    }
}
